models/linear_model.py

```diff
@@ -1,28 +1,69 @@
-from .model_base import Model
+from model_base import Model
 import numpy as np
 
 
 class LinearRegressionModel(Model):
     """
     A model for Linear Regression.
-    TODO - Implement!
+    Arguments:
+        input_features (int)
+        output_features (int)
     """
+
+    def __init__(self, input_features, output_features):
+        self.input_features = input_features
+        self.output_features = output_features
+        self.weights = np.zeros((input_features + 1, output_features))
+
+    @staticmethod
+    def design_matrix(X):
+        """Take dataset X and return the design-matrix (dm_X)"""
+        return np.c_[np.ones(X.shape[0]), X]
+
     def predict(self, X):
-        pass
+        """Predict the output after training by given dataset X"""
+        return self.design_matrix(X) @ self.weights
 
-    def fit(self, X, y, epochs=None):
-        pass
+    def fit(self, X, Y, epochs=None, learn_rate=None):
+        """
+        Training the model by dataset X and the true values of Y.
+        Arguments:
+            X (array): Dataset.
+            Y (array): True values.
+            Argument for 'Gradient Descent' method:
+                epochs (int): Num of iteration on GD function.
+                learn_rate (float): The rate of the learning of the model.
+        """
+        dm_X = self.design_matrix(X)
 
-    def loss(self, y_prediction, y_true):
+        # 'Gradient Descent' method if learn_rate and epochs was given
+        if learn_rate and epochs:
+            for i in range(epochs):
+                self.weights = self.gd(dm_X, Y, learn_rate)
+
+        # 'Normal equation' method if learn_rate and epochs wasn't given
+        else:
+            self.weights = np.linalg.inv(dm_X.T @ dm_X) @ dm_X.T @ Y
+
+    def gd(self, dm_X, Y, learn_rate):
+        """Apply one step in the descent of the Loss function"""
+        new_weights = np.copy(self.weights)
+        for (i, j), _ in np.ndenumerate(new_weights):
+            new_weights[i, j] -= learn_rate * np.mean(
+                (np.squeeze(np.asarray(dm_X @ self.weights[:, j])) - Y[:, j]) * dm_X[:, i])
```

Comment:
I like `np.squeeze(dm_X @ self.weights[:, j])`, this is a great idea. `np.asarray` is probably unnecessary.